### PR TITLE
Homework2: Alexander Zhuk

### PR DESCRIPTION
**Pull request by ZhukAlexander90**
Домашняя работа №2 в отдельной ветке

### CODE REVIEW

File: Homework2_ZhukAlexander/task1.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# Считает общую цену. Вводится: М-рубли, N-копейки
+# L-количество товара
+
+M = int(input("Введите цену в рублях: "))
+N = int(input("Введите цену в копейках: "))
+L = int(input("Количество предметов: "))
+rub = M * L
+cent = N * L
+if cent > 100:                      # если больше 100 копеек
```
Comment (belovmd) [inline Homework2_ZhukAlexander/task1.py]: redundant comment

File: Homework2_ZhukAlexander/task2_easy_unpack.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+#  to the last for the given array.
+
+
+def easy_unpack(elements: tuple) -> tuple:
+    """returns a tuple with 3 elements - first, third and second to the last"""
+    first_elem = elements[0]
+    second_elem = elements[2]
+    pre_last = elements[-2]
+    new_tuple = (first_elem, second_elem, pre_last)
+    return new_tuple
```
Comment (belovmd) [inline Homework2_ZhukAlexander/task2_easy_unpack.py]: just
`return elements[0], elements[2] etc`

File: Homework2_ZhukAlexander/task3.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Найти самое длинное слово в предложении
+
+import re
+newStr = input("Предложение: ")
+newStr.strip()
+newList = re.split(' |, |: |; |-|\n', newStr)
```
Comment (belovmd) [inline Homework2_ZhukAlexander/task3.py]: (){}[] etc?

File: Homework2_ZhukAlexander/task3.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+# Найти самое длинное слово в предложении
+
+import re
+newStr = input("Предложение: ")
+newStr.strip()
```
Comment (belovmd) [inline Homework2_ZhukAlexander/task3.py]: bad naming - we use snake case according to PEP8

File: Homework2_ZhukAlexander/task4.py [hunk 5]
```diff
@@ -0,0 +1,9 @@
+# Требуется удалить повторяющиеся символы и пробелы
+
+enterString = input("Введите строку: ")
```
Comment (belovmd) [inline Homework2_ZhukAlexander/task4.py]: bad naming - camel case

File: Homework2_ZhukAlexander/task5.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
+# Считаем количество строчных и прописных букв
+# вариант первый
+
+enterString = input("Введите строку (англ. символы): ")
+upLet = 0
```
Comment (belovmd) [inline Homework2_ZhukAlexander/task5.py:21]: bad naming

File: Homework2_ZhukAlexander/task5.py [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+lowLet = 0
+upLetters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'
+lowLetters = 'abcdefghijklmnopqrstuvwxyz'
+for letter in enterString:
```
Comment (belovmd) [inline Homework2_ZhukAlexander/task5.py]: islower(), isupper() string methods?

File: Homework2_ZhukAlexander/task6_Fibon.py [hunk 8]
```diff
@@ -0,0 +1,9 @@
+# Выводит N-ое число Фибоначчи
+
+num = int(input("Введите N: "))
+earlyNum = nextNum = 1      # т.к. 1й и 2й элементы посл-ти == 1
+for n in range(2, num):
+    temp = earlyNum + nextNum
+    earlyNum = nextNum
```
Comment (belovmd) [inline Homework2_ZhukAlexander/task6_Fibon.py:7]: a, b = b, a+b ?

File: Homework2_ZhukAlexander/task6_palindromic.py [hunk 9]
```diff
@@ -0,0 +1,13 @@
+# Определяет, является ли введенное число
+# палиндромом
+
+num = int(input("Введите N: "))
+oldNum = num
+newNum = 0
+while oldNum > 0:
```
Comment (belovmd) [inline Homework2_ZhukAlexander/task6_palindromic.py]: 👍

File: Homework2_ZhukAlexander/task6_palindromic.py [hunk 10]
```diff
@@ -0,0 +1,13 @@
+# Определяет, является ли введенное число
+# палиндромом
+
+num = int(input("Введите N: "))
+oldNum = num
```
Comment (belovmd) [inline Homework2_ZhukAlexander/task6_palindromic.py]: bad naming

**Review by belovmd — COMMENTED** [review]
1. Please remove tasks from first homework
2. Please rename Pull Request - it contradicts to our guidelines
3. It seems tasks from external tasks were pretty easy. you can add some more advanced tasks
4. please remove tasks from lessons
5. Please address next issues:

File: Homework2_ZhukAlexander/task1.py (hunk 1 above)
Comment (ZhukAlexander90) [reply Homework2_ZhukAlexander/task1.py]: > 1. Please remove tasks from first homework
> 2. Please rename Pull Request - it contradicts to our guidelines
> 3. It seems tasks from external tasks were pretty easy. you can add some more advanced tasks
> 4. please remove tasks from lessons
> 5. Please address next issues:

put in order the repository. Homework2 without tasks HW1 and lessons
renamed

**Comment by ZhukAlexander90** [comment]
> 1. Please remove tasks from first homework
> 2. Please rename Pull Request - it contradicts to our guidelines
> 3. It seems tasks from external tasks were pretty easy. you can add some more advanced tasks
> 4. please remove tasks from lessons
> 5. Please address next issues:


put in order the repository. Homework2 without tasks HW1 and lessons
renamed

File: Homework2_ZhukAlexander/task1.py (hunk 1 above)
Comment (ZhukAlexander90) [reply Homework2_ZhukAlexander/task1.py]: done

File: Homework2_ZhukAlexander/task3.py (hunk 3 above)
Comment (ZhukAlexander90) [reply Homework2_ZhukAlexander/task3.py]: made with import string

File: Homework2_ZhukAlexander/task3.py (hunk 4 above)
Comment (ZhukAlexander90) [reply Homework2_ZhukAlexander/task3.py]: done

File: Homework2_ZhukAlexander/task4.py (hunk 5 above)
Comment (ZhukAlexander90) [reply Homework2_ZhukAlexander/task4.py]: done

File: Homework2_ZhukAlexander/task5.py (hunk 6 above)
Comment (ZhukAlexander90) [reply Homework2_ZhukAlexander/task5.py:21]: done

File: Homework2_ZhukAlexander/task5.py (hunk 7 above)
Comment (ZhukAlexander90) [reply Homework2_ZhukAlexander/task5.py]: add 3rd version with islower(), isupper()

File: Homework2_ZhukAlexander/task6_palindromic.py (hunk 10 above)
Comment (ZhukAlexander90) [reply Homework2_ZhukAlexander/task6_palindromic.py]: done

File: Homework2_ZhukAlexander/task5.py [hunk 11]
```diff
@@ -0,0 +1,45 @@
...
+        low_let += 1
+    else:
+        continue
+print('Строчных букв: ', low_let)
+print('Прописных букв: ', up_let)
+
+# вариант второй
+enter_string = input("Введите строку (англ. символы): ")
+upLet = 0
+lowLet = 0
```
Comment (belovmd) [inline Homework2_ZhukAlexander/task5.py:22]: not python naming

**Review by belovmd — COMMENTED** [review]
8,5

**Pull request closed without merging**